book/types_of_data.md

```diff
@@ -395,11 +396,30 @@ To remove rows from a table, you'll commonly use the `select` command with row n
 
 There are numerous other commands for selecting and reducing the data in tables, records and lists.
 
+## Closures
+
+Closures are anonymous functions that can close over variables outside their scope and can be passed as a value.
+For example, in the command `each { |it| print $it }` the closure is the portion contained in curly braces, `{ |it| print $it }`.
+Closure parameters are specified between a pair of pipe symbols (for example, `|it|`) if necessary.
+You can also use `$in` in most closures instead of providing a parameter: `each { print $in }`
+
+Closures are a useful way to represent code that can be executed on each row of data.
+It is idiomatic to use `$it` as a parameter name in [`each`](commands/each.md) blocks, but not required;
```

Comment:
Love the crossreferencing!

Reply:
Unfortunately, not my work.
I just copied the block paragraph and replaced block with closure :P

Reply:
Thanks for keeping it around then :)